src/modm/board/devebox_stm32h750vb/board.hpp

```diff
@@ -0,0 +1,172 @@
+	static constexpr uint32_t I2c4 = Apb4;
+
+	static constexpr uint32_t Apb1Timer = Apb1 * 2;
+	static constexpr uint32_t Apb2Timer = Apb2 * 2;
+	static constexpr uint32_t Timer1  = Apb2Timer;
+	static constexpr uint32_t Timer2  = Apb1Timer;
+	static constexpr uint32_t Timer3  = Apb1Timer;
+	static constexpr uint32_t Timer4  = Apb1Timer;
+	static constexpr uint32_t Timer5  = Apb1Timer;
+	static constexpr uint32_t Timer6  = Apb1Timer;
+	static constexpr uint32_t Timer7  = Apb1Timer;
+	static constexpr uint32_t Timer8  = Apb2Timer;
+	static constexpr uint32_t Timer12 = Apb1Timer;
+	static constexpr uint32_t Timer13 = Apb1Timer;
+	static constexpr uint32_t Timer14 = Apb1Timer;
+	static constexpr uint32_t Timer15 = Apb2Timer;
+	static constexpr uint32_t Timer16 = Apb2Timer;
+	static constexpr uint32_t Timer17 = Apb2Timer;
+
+	static constexpr uint32_t Usb = 50_MHz; // From PLL3Q
+
+	static bool inline
+	enable()
+	{
+		Rcc::enableExternalCrystal(); // 25MHz
+		Rcc::setVoltageScaling(Rcc::VoltageScaling::Scale0); // required for 400MHz/480MHz
+		const Rcc::PllFactors pllFactors1{
+			.range = Rcc::PllInputRange::MHz8_16,
+			.pllM = 2,		//   25MHz / M= 12.5MHz
+			.pllN = 64,		// 12.5MHz * N=  800MHz
+			.pllP = 2,		//  800MHz / P=  400MHz = F_cpu
+			.pllQ = 20,		//  800MHz / Q=   40MHz
+			.pllR = 20,		//  800MHz / R=   40MHz
+		};
+		Rcc::enablePll1(Rcc::PllSource::ExternalClock, pllFactors1);
+		// Use PLL3 for USB 50MHz
+		const Rcc::PllFactors pllFactors3{
+			.range = Rcc::PllInputRange::MHz8_16,
+			.pllM = 5,		//   25MHz / M=     5MHz
+			.pllN = 48,		//    5MHz * N=   240MHz
+			.pllP = 2,		//  240MHz / P=   120MHz
+			.pllQ = 5,		//  240MHz / Q=    48MHz = F_usb
+			.pllR = 5,		//  240MHz / R=    48MHz
+		};
+		Rcc::enablePll3(Rcc::PllSource::ExternalClock, pllFactors3);
+		Rcc::setFlashLatency<Ahb>();
+		// max 240MHz on AHB
+		Rcc::setAhbPrescaler(Rcc::AhbPrescaler::Div2);
+		// max 120MHz on APB
+		Rcc::setApb1Prescaler(Rcc::Apb1Prescaler::Div2);
+		Rcc::setApb2Prescaler(Rcc::Apb2Prescaler::Div2);
+		Rcc::setApb3Prescaler(Rcc::Apb3Prescaler::Div2);
+		Rcc::setApb4Prescaler(Rcc::Apb4Prescaler::Div2);
+		// update clock frequencies
+		Rcc::updateCoreFrequency<Hclk>();
+		Rcc::enableUsbClockSource(Rcc::UsbClockSource::Pll3Q);
+		// Switch the main clock source to PLL
+		Rcc::enableSystemClock(Rcc::SystemClockSource::Pll1P);
+
+		return true;
+	}
+};
+
+
+using ButtonK1 = GpioInverted<GpioInputE3>;
+using ButtonK2 = GpioInverted<GpioInputC5>;
+
+using LedGreen = GpioInverted<GpioOutputA1>;	// User LED
+
+using Leds = SoftwareGpioPort< LedGreen >;
+
+namespace usb
+{
+using Dm = GpioA11;		// DM: USB_DM
+using Dp = GpioA12;		// DP: USB_DP
+}
+
+
+inline void
+initialize()
+{
+	SystemClock::enable();
+	SysTickTimer::initialize<SystemClock>();
+
+	LedGreen::setOutput(modm::Gpio::Low);
+
+	ButtonK1::setInput(Gpio::InputType::PullUp);
+	ButtonK1::setInputTrigger(Gpio::InputTrigger::FallingEdge);
+	ButtonK1::enableExternalInterrupt();
```

Comment:
Sure?

Reply:
This just sets the flags, so you can lazily poll if there has been an input. We don't use it anywhere, we could also remove it from all BSPs.